pkg/controller/managedcluster/managedcluster_controller.go

```diff
@@ -232,6 +236,24 @@ func (r *ReconcileManagedCluster) deleteManagedClusterNamespace(
 	return nil
 }
 
+func (r *ReconcileManagedCluster) getClusterName(ctx context.Context, cluster *clusterv1.ManagedCluster) (string, error) {
+	clusterDeployment := &hivev1.ClusterDeployment{}
+	err := r.client.Get(ctx, types.NamespacedName{Namespace: cluster.Name, Name: cluster.Name}, clusterDeployment)
+	if errors.IsNotFound(err) {
+		return cluster.Name, nil
+	}
+
+	if err != nil {
+		return "", err
+	}
+
+	if clusterDeployment.Spec.ClusterPoolRef != nil {
```

Comment:
You should also need check if clusterDeployment.Spec.ClusterPoolRef.ClaimName exist.